pkg/ccl/backupccl/testdata/backup-restore/descriptor-conflicts

```diff
@@ -153,4 +135,3 @@ should survive
 query-sql
 select role_name, settings FROM system.database_role_settings AS drs JOIN system.namespace AS ns ON ns.id = drs.database_id WHERE ns.name = 'foo';
 ----
-hamburger {application_name=helper}
```

Comment:
hmm why don't we restore this anymore?

Reply:
Ha, nice catch -- there was a bug in the prior PR that this was pointing to that was previously hidden by the fact we remapped 1:1. Fixed with new first commit here.

Reply:
er, by first I meant 5th: [backupccl: remap all IDs in one pass](https://github.com/cockroachdb/cockroach/pull/85492/commits/6c54e9a8e6430c0f3b496d0909ae86f10920d50d)